### DISCUSSION TITLE
Recommended version of EZSP firmware

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1268] [Q&A]
in the link here after, they are some recommendation of minimum firmware level, but also it is mentioned that that 7.x are still in beta.

https://github.com/zigpy/zigpy/wiki/Coordinator-Firmware-Updates#recommended-firmware-versions-for-ezsp-coordinators

However Sonoff ZBDongle-E can be upgraded to 7.3.1.0

**Comment by MattWestb** [comment]
MG1X devices like my IKEA Billy is only supported until NCP / EZSP 6.X and the stable for all of them is 6.7.8.0 and candidate is 6.10.X that is the last possible bug fixed version then using NCP firmware and looks working great but luck of user feed back for making it flagged as stable. 

The same recommendations is for MG12 and MG2X chips plus latest 7.3.X firmware but nothing between then its having too many bugs and not all new futures implanted that is needed for backup and restore the chip.
6.8.X and 6.9.X is not banned as all as 7.1.X and 7.2 (all is more or less alpha / beta firmware in the silabs code).

All can also running RCP firmware that is on the latest 7.3.X firmware (or latest tested with the addon) and is working great with ZHA and some Z2M users system.

Also 7.3.X is implanting Zigbee PRO 2023 / R23 that is the upcoming merger of Zigbee and matter ZCL and modern security for the mesh network.

More testing and reporting is welcome and also updating the wiki pages with more relevant information.

PS: 7.1.X can having problems with NVM storage if updating to other version and going back and need getting it being re flashed for working = only problems.

**Comment by Hedda** [comment]
> PS: 7.1.X can having problems with NVM storage if updating to other version and going back and need getting it being re flashed for working = only problems.

FYI, I believe that Nabu Casa is still promoting EZSP 7.1.1.0 NCP firmware as stable for the Home Assistant SkyConnect USB dongle, or?

https://github.com/NabuCasa/silabs-firmware/tree/main/EmberZNet

EZSP 7.3.x.x is available under the beta directory:

https://github.com/NabuCasa/silabs-firmware/tree/main/EmberZNet/beta

**Reply by MattWestb** [reply]
RCP is running on the latest (wrong then the latest is not being added to the addon) but 4/7.3.X is the most good working ones for MG2X devices then the earlier have some nasty bugs and cant using soft IEEEE for the chip.
